### DISCUSSION TITLE
Would it be possible to track “entry” & “exit” pages?

### DISCUSSION THREAD

**Discussion by JaneJeon** [umami-software/umami#611] [Q&A]
Hi Mike, absolutely loving the work you’re doing.

I know that a goal of the project is to track only what’s needed, but would it not be possible to track “entry” and “exit” pages without a cookie? Theoretically, it might be possible by adding query parameters to each link (akin to tracking request id) so that you know when someone has come from another link in the site, and more importantly, when someone *left* the site after visiting a page.

I know you must be burdened with maintenance, but I just wanted to throw this idea out there!

**Comment by mikecao** [comment]
Not sure what you mean by entry. All page visits have a referrer which tells you where the user came from. For exits, we need to update the tracker logic which is covered in #446.

**Comment by JaneJeon** [comment]
Sorry, I should’ve been clearer. Yes, I did mean referrer by “entry”. But I was talking more about tracking what was the *last* page in the session that the user visited before clicking off.
